addons/web_hierarchy/static/src/hierarchy_model.js

```diff
@@ -670,29 +661,56 @@ export class HierarchyModel extends Model {
         config.context = "context" in params ? params.context : config.context;
         if ("domain" in params) {
             config.domain = params.domain;
-            if (!params.domain.length) {
-                if (config.context.hierarchy_res_id) {
-                    config.domain = [["id", "=", config.context.hierarchy_res_id]];
-                    delete config.context.hierarchy_res_id; // just needed for the first load.
-                } else {
-                    config.domain = this.defaultDomain;
-                }
+            if (!this.env.searchModel?.query.length && config.context.hierarchy_res_id) {
+                config.domain = [["id", "=", config.context.hierarchy_res_id]];
```

Comment:
@xavierbol Can you double check that my modification to the application of `hierarchy_res_id` seems correct to you? I'm not really sure about the specifics :)